src/shipit_code_coverage/shipit_code_coverage/codecov.py

```diff
@@ -145,30 +142,48 @@ def generate_info(self, commit_sha, coveralls_token, suite=None):
             if suite is None or suite in fname:
                 ordered_files.append('ccov-artifacts/' + fname)
 
-        r = requests.get('https://hg.mozilla.org/mozilla-central/json-rev/%s' % self.revision)
-        push_id = r.json()['pushid']
-
         cmd = [
           'grcov',
-          '-t', 'coveralls',
+          '-t', out_format,
           '-s', self.repo_dir,
           '-p', '/home/worker/workspace/build/src/',
           '--ignore-dir', 'gcc',
           '--ignore-not-existing',
-          '--service-name', 'TaskCluster',
-          '--service-number', str(push_id),
-          '--commit-sha', commit_sha,
-          '--token', coveralls_token,
         ]
 
-        if suite is not None:
-            cmd.extend(['--service-job-number', str(self.suites.index(suite) + 1)])
-        else:
-            cmd.extend(['--service-job-number', '1'])
+        if out_format == 'coveralls':
+            r = requests.get('https://hg.mozilla.org/mozilla-central/json-rev/%s' % self.revision)
```

Comment:
Maybe check the status code
```
r.raise_for_status()
```
or usually i do
```
assert r.ok, "Failed to retrieve mozilla revision {}".format(self.revision)
```

Reply:
Good idea, I'm calling `raise_for_status` now.